apps/api-extractor/src/analyzer/PackageMetadataManager.ts

```diff
@@ -61,11 +61,34 @@ export class PackageMetadataManager {
   private readonly _packageMetadataByPackageJsonPath: Map<string, PackageMetadata>
     = new Map<string, PackageMetadata>();
 
-  public static writeTsdocMetadataFile(packageJsonFolder: string): void {
-    // This feature is still being standardized: https://github.com/Microsoft/tsdoc/issues/7
-    // In the future we will use the @microsoft/tsdoc library to read this file.
-    const tsdocMetadataPath: string = path.join(packageJsonFolder,
-      'dist', PackageMetadataManager.tsdocMetadataFilename);
+  // This feature is still being standardized: https://github.com/Microsoft/tsdoc/issues/7
+  // In the future we will use the @microsoft/tsdoc library to read this file.
+  private static _getTsdocMetadataPathFromPackageJson(packageFolder: string, packageJson: IPackageJson): string {
+    const { tsdocMetadataFilename } = PackageMetadataManager;
+    let tsdocMetadataRelativePath: string;
+    if (packageJson.tsdocMetadata) {
+      tsdocMetadataRelativePath = packageJson.tsdocMetadata;
+    } else if (packageJson.typings) {
+      tsdocMetadataRelativePath = path.join(path.dirname(packageJson.typings), tsdocMetadataFilename);
+    } else if (packageJson.main) {
+      tsdocMetadataRelativePath = path.join(path.dirname(packageJson.main), tsdocMetadataFilename);
+    } else {
+      tsdocMetadataRelativePath = tsdocMetadataFilename;
+    }
+    const tsdocMetadataPath: string = path.join(packageFolder, tsdocMetadataRelativePath);
+    return tsdocMetadataPath;
+  }
+
+  public static writeTsdocMetadataFile(packageFolder: string, packageJson: IPackageJson, outputPath?: string): void {
```

Comment:
>writeTsdocMetadataFile(packageFolder: string, packageJson: IPackageJson, outputPath?: string): void { [](start = 16, length = 101)

I found this API a little confusing.  (What's the difference between packageFolder and outputPath?  Why is outputPath optional? etc)

It might be simpler to have a function that figures out the path:

```
/**
 * @param tsdocMetadataPath - An explicit path that can be configured in api-extractor.json.
 * If this parameter is not an empty string, it overrides the normal path calculation.
 * @returns the absolute path to the TSDoc metadata file
 */
resolveTsdocMetadataPath(packageFolder, packageJson, tsdocMetadataPath: string): string;

/**
 * Writes the TSDoc metadata file to the specified output file.
 */
writeTsdocMetadataFile(tsdocMetadataPath: string): void.
```

Then you could test these functions independently.